--- a/UI/Qt/CMakeLists.txt
+++ b/UI/Qt/CMakeLists.txt
@@ -44,7 +44,7 @@
     WindowControlButton.cpp
     ladybird.qrc
 )
-target_link_libraries(ladybird PRIVATE Qt::Core Qt::Gui Qt::Widgets)
+target_link_libraries(ladybird PRIVATE Qt::Core Qt::Gui Qt::Widgets -Wl,--whole-archive harfbuzz fontconfig -Wl,--no-whole-archive)
 
 if (TARGET Qt6::Positioning)
     target_sources(ladybird PRIVATE GeolocationProviderQt.cpp GeolocationProviderQt.h)
